compiler/modules/front.syntax/parseStmt.bal

```diff
@@ -222,14 +225,39 @@ function parseCompoundAssignStmt(Tokenizer tok, LExpr lValue, CompoundAssignOp o
     return stmt; 
 }
 
+function parseDestructuringAssignStmt(Tokenizer tok) returns DestructuringAssignStmt|err:Syntax {
+    BindingPattern pattern = check parseBindingPattern(tok);
+    check tok.expect("=");
+    Expr expr = check parseExpr(tok);
+    DestructuringAssignStmt stmt = { pattern, expr };
+    check tok.expect(";");
+    return stmt;
+}
+
+function parseBindingPattern(Tokenizer tok) returns BindingPattern|err:Syntax {
+    Token? cur = tok.current();
+    check tok.advance();
+    if (cur == "_") {
+        return WildcardBindingPattern;
+    }
+    return parseError(tok, "unsupported binding pattern"); // TODO: check the message
+}
+
 function parseVarDeclStmt(Tokenizer tok, boolean isFinal = false) returns VarDeclStmt|err:Syntax {
     TypeDesc td = check parseInlineTypeDesc(tok);
     return finishVarDeclStmt(tok, td, isFinal);
 }
 
 function finishVarDeclStmt(Tokenizer tok, TypeDesc td, boolean isFinal = false) returns VarDeclStmt|err:Syntax {
     Token? cur = tok.current();
-    string varName = check tok.expectIdentifier();
+    string varName;
```

Comment:
This is not right. Do assignments first. Then do var decls as a separate pull request.

Reply:
Removed